clients/src/main/java/org/apache/kafka/common/security/kerberos/KerberosShortNamer.java

```diff
@@ -95,7 +95,12 @@ public String shortName(KerberosName kerberosName) throws IOException {
             if (result != null)
                 return result;
         }
-        throw new NoMatchingRule("No rules applied to " + toString());
+        throw new NoMatchingRule("No rules " + principalToLocalRules + " applied to " + kerberosName);
```

Comment:
Nitpick: maybe the rules should be at the end since they could be long. Something like:

```java
throw new NoMatchingRule("No rule applies to " + kerberosName + ", rules " + principalToLocalRules);
```